### ISSUE TITLE
`example` not picked up if coming from an `anyOf`

### ISSUE THREAD

**Issue by Corwinpro** [mockoon/mockoon#948]
**Describe the bug**

If a `schema` is described via the `allOf`, the `example` values are not properly taken into account when generating the mock schema.

Please see the section below for more details.

**To Reproduce**

Tested from the latest `main` branch, building and running as described in the contribution guide.

Example `openapi.yaml`:

```yaml
openapi: 3.0.1
info:
  version: 1.0.0

components:
  schemas:
    Model:
      type: object
      properties:
        name:
          type: string
      example:
        name: custom_name

paths:
  /model:
    get:
      summary: Returns a model.
      description: Some description.
      responses:
        '200':
          description: A model with a name
          content:
            application/json:
              schema:
                allOf:
                  - $ref: '#/components/schemas/Model'
```

Now running the server:

```sh
./packages/cli/bin/run start -D --data /PATH/TO/openapi.yaml --port 3000
```

and querying the endpoint:

```
$ curl -X GET localhost:3000/model
{
  "name": ""  # This should not be empty. Expected "custom_name"
}
```

**Mockoon version:**

Latest `main` (1.22.dev?)

**OS / OS version:**

Mac OS M1 ARM

### How to fix

I believe the problem is a subtle interaction of this

https://github.com/mockoon/mockoon/blob/d943b0208b4608eca194efb9b415660dc291ef17/packages/commons-server/src/libs/openapi-converter.ts#L503-L512

and this

https://github.com/mockoon/mockoon/blob/d943b0208b4608eca194efb9b415660dc291ef17/packages/commons-server/src/libs/openapi-converter.ts#L464

When going through the `allOf` logical path, the `typeFactory` would be `object` (for the example described above) and the `schemaToBuild` is

```js
{
  type: 'object',
  properties: { name: { type: 'string' } },
  example: { name: 'custom_name' }
}
```

But the factory corresponding to `object` does not take into account the `example` field (which is wrong). So the `object` factory should be updated like

```js
      object: (objectSchema) => {
        if (objectSchema.example) {
          return objectSchema.example;
        }

        ...
```

In fact, maybe all three "special" cases

https://github.com/mockoon/mockoon/blob/d943b0208b4608eca194efb9b415660dc291ef17/packages/commons-server/src/libs/openapi-converter.ts#L486-L499

should be refactored into a helper function `handle_schema_defaults(schema)`, and the `object` factory should use it (not just the `objectSchema.example` check).

I tested the fix as described in the snippet above and it works.

Thank you!

**Comment by Corwinpro** [comment]
UPD: I think the issue is actually with the `allOf`/`anyOf`/`oneOf` code specifically, not how the `object` factory works.

I think the following code:

https://github.com/mockoon/mockoon/blob/d943b0208b4608eca194efb9b415660dc291ef17/packages/commons-server/src/libs/openapi-converter.ts#L503-L512

Should be replaced with

```js
  // check if we have an array of schemas, and take first item
  for (const propertyName of ['allOf', 'oneOf', 'anyOf']) {
    if (
      schema.hasOwnProperty(propertyName) &&
      schema[propertyName].length > 0
    ) {
      return generateSchema(schema[propertyName][0]);
    }
  }
```

**Comment by 255kb** [comment]
Thank you for reporting. 
I guess you are right, your issue seems ot be linked to the fact that we are only taking the first element present in any `allOf`, `oneOf`, `anyOf`.

Someone is currently working on supporting `example` and not only `schema`: https://github.com/mockoon/mockoon/issues/962
This will update the parser so it creates as many responses as we have schemas and examples. We should probably do the same when encountering `allOf`, `oneOf`, `anyOf`.

**Comment by Corwinpro** [comment]
@255kb Thank you! Really appreciate you looking into this.

> your issue seems ot be linked to the fact that we are only taking the first element present in any allOf, oneOf, anyOf.

To clarify this (I apologise that my original solution sketch was a bit confusing): the problem is not the first elements being picked up. The issue is that whatever we pick up in the all/oneof/anyof does not go through the standard schema generation pipeline but gets passed directly into `object` generation.

> Someone is currently working on supporting example and not only schema: https://github.com/mockoon/mockoon/issues/962

Awesome, I hope that work is finished and merged. I am not entirely sure though that is adderessing the problem described in this PR.

**Comment by 255kb** [comment]
📦 This is now fixed in [v6.2.0](https://mockoon.com/releases/6.2.0/) using your code sample @Corwinpro 
I credited you in the changelog. Thanks!
